malloc_free/bpf/malloc_free.bpf.c

```diff
@@ -44,10 +44,15 @@ struct malloc_event _malloc_event = {};
 
 char LICENSE[] SEC("license") = "Dual BSD/GPL";
 
+// Configuration variables set from userspace
+u32 max_events = 8192;
+u32 max_records = 1024;
+u32 max_stack_depth = PERF_MAX_STACK_DEPTH;
+
 // Hash map to store malloc event records
 struct {
 	__uint(type, BPF_MAP_TYPE_HASH);
-	__uint(max_entries, 8192);
+	__uint(max_entries, 16384); // Will be configured from userspace
```

Comment:
how is this one configured?

Reply:
This can be configured in user space code like following:

```
    open_skel
        .maps_mut()
        .malloc_event_records()
        .set_max_entries(cli.max_events)
        .map_err(|_| Report::new(JtraceError::BPFError))
        .attach_printable("Failed to set malloc_event_records max_entries")?;

```